src/opm/input/eclipse/Schedule/Network/NetworkKeywordHandlers.cpp

```diff
@@ -170,6 +172,29 @@ void handleNODEPROP(HandlerContext& handlerContext)
         }
 
         if (as_choke) {
+            const auto& group = handlerContext.state().groups.get(name);
+            if (group.wellgroup()) {
+                //Wells belong to a group with autochoke enabled are to be run on a common THP and should not have guide rates
+                for (const std::string& wellName : group.wells()) {
+                    auto well = handlerContext.state().wells.get(wellName);
+
+                    //Let the wells be operating on a THP Constraint
+                    auto properties = std::make_shared<Well::WellProductionProperties>(well.getProductionProperties());
+                    //The wells are not to be under GRUP control using guide rates but under THP control
+                    properties->dropProductionControl(Well::ProducerCMode::GRUP);
+                    properties->addProductionControl(Well::ProducerCMode::THP);
+                    properties->controlMode = Well::ProducerCMode::THP;
+                    well.updateProduction(properties);
+
+                    // Guide rate availability should be set to false
+                    well.updateAvailableForGroupControl(false);
+                    auto new_config = handlerContext.state().guide_rate();
+                    new_config.update_well(well);
+                    handlerContext.state().guide_rate.update( std::move(new_config) );
+                    handlerContext.state().wells.update( std::move(well) );
+                }
+            }
```

Comment:
because only **well group** can be autochoke node, I believe the code below until `node.as_choke(target_group);` should be included in the if condition
```c++
if (group.wellgroup()) {
```

Reply:
The current implementation for auto-choke allows indeed only well groups. I have added a throw in case auto-choke is set for a group without wells with error message: _The auto-choke option is implemented only for well groups_.

Reply:
Done

Reply:
Referring to our chat we keep the error message.